pep-9999.rst

```diff
@@ -0,0 +1,226 @@
+This PEP adds data which were previously only available through the JSON API,
+in order to allow more clients which were previously Warehouse specific to
+support arbitrary standards-compliant indexes.
+
+
+Specification
+=============
+
+This specification defines version 1.2 of the simple repository API. For the
+HTML version of the API, there is no change from version 1.0. For the JSON
+version of the API, the following changes are made:
+
+- The ``api-version`` must specify version 1.2 or later.
+- A new ``vulnerabilities`` key is added at the top level.
+- One new "file information" key, ``vulnerability-ids`` is added to the
+  ``files`` data.
+
+The ``vulnerabilties`` and ``vulnerability-ids`` keys are mandatory.
+
+Vulnerabilities
+---------------
+
+An additional key, ``vulnerabilities`` MUST be present at the top level, in
+addition to the keys previous defined in :pep:`691` and :pep:`700`. This key
+MUST contain a mapping from a unique vulnerability identifier to a dictionary
+of fields that describe that vulnerability, for every vulnerability that
+affects the given project.
+
+An example of the structure for the ``vulnerabilities`` field:
+
+.. code-block:: json
+
+   "vulnerabilities": {
+       "PYSEC-2022-XXX": {
+           "aliases": [
+               "CVE-2022-XXXXX"
+           ],
+           "details": "A long description.",
+           "summary": "A shorter summary.",
+           "fixed_in": [
+               "1.2.3"
+           ],
+           "withdrawn": "2022-06-28T16:39:06Z"
+       }
+   }
+
+
+For this field, the key (vulnerability ID) MUST be unique, and MUST take the
+form of the `OSV ID field
+<https://ossf.github.io/osv-schema/#id-modified-fields>`__. If a ``PYSEC-``
+prefixed identifier is available for a given vulnerability, this ID should be
+preferred as the primary vulnerability identifier, and other identifiers should
+be added to the ``aliases`` field.
+
+The dictionary corresponding to that key MUST take the form of the `Open Source
+Vulnerabilty format (v. 1.3.1) <https://ossf.github.io/osv-schema/>`__.
+Specifically, the following fields MUST be included:
+
+- ``aliases``: `A list of IDs of the same vulnerability in other databases <https://ossf.github.io/osv-schema/#aliases-field>`__.
+- ``details``: `Additional English textual details about the vulnerability, formatted with CommonMark <https://ossf.github.io/osv-schema/#summary-details-fields>`__.
+- ``summary``: `A one-line, English textual summary of the vulnerability, with no markup <https://ossf.github.io/osv-schema/#summary-details-fields>`__.
+- ``withdrawn`` `The time the entry should be considered to have been withdrawn <https://ossf.github.io/osv-schema/#withdrawn-field>`__.
+
+All other fields are optional. All fields that return lists MUST return empty
+lists if no value is present. All fields that return strings MUST return
+``null`` if no value is present.
+
+Additionally, a field ``fixed_in`` MUST be included. This field represents a
```

Comment:
Could this go in the list of required fields above? We make it seem like a comprehensive list of required fields so it's a bit confusing that a few paragraphs below, we introduce the mandatory `fixed_in` field.

Reply:
Those were supposed to be required fields from the schema (`fixed_in` is not in the schema) but I can see how that would be unclear.

Would separating this into two sections, "Required schema fields" and "Additional non-schema fields", or somehting similar, make sense?

Reply:
Ah I see what you meant.

That seems reasonable to me.